brage-import/src/main/java/no/sikt/nva/brage/migration/merger/publicationinstancemerger/PublicationInstanceMerger.java

```diff
@@ -2,20 +2,71 @@
 
 import static java.util.Objects.nonNull;
 import no.unit.nva.model.PublicationDate;
+import no.unit.nva.model.instancetypes.PublicationInstance;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.degree.DegreeLicentiate;
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.instancetypes.degree.OtherStudentWork;
+import no.unit.nva.model.instancetypes.journal.AcademicArticle;
+import no.unit.nva.model.instancetypes.journal.JournalIssue;
+import no.unit.nva.model.instancetypes.journal.JournalLeader;
+import no.unit.nva.model.instancetypes.journal.ProfessionalArticle;
+import no.unit.nva.model.instancetypes.media.MediaFeatureArticle;
+import no.unit.nva.model.instancetypes.report.ConferenceReport;
+import no.unit.nva.model.instancetypes.report.ReportBasic;
+import no.unit.nva.model.instancetypes.report.ReportBookOfAbstract;
+import no.unit.nva.model.instancetypes.report.ReportResearch;
+import no.unit.nva.model.instancetypes.report.ReportWorkingPaper;
 import no.unit.nva.model.pages.MonographPages;
+import no.unit.nva.model.pages.Range;
 import nva.commons.core.JacocoGenerated;
 
-public class PublicationInstanceMerger {
+public abstract class PublicationInstanceMerger<T extends PublicationInstance<?>> {
```

Comment:
This inheritance pattern is a bit weird. Ask yourself if it could be solved with interfaces instead, and if yes, do that instead. 

Do we need to inherit the PublicaitonInstance properties and methods here?